src/Analyzer.Utilities/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -93,8 +93,18 @@ public static IEnumerable<ITypeSymbol> GetBaseTypesAndThis(this ITypeSymbol type
             }
         }
 
-        public static bool DerivesFrom(this INamedTypeSymbol symbol, INamedTypeSymbol candidateBaseType)
+        public static bool DerivesFrom(this ITypeSymbol symbol, ITypeSymbol candidateBaseType, bool baseTypesOnly = false)
```

Comment:
Will file. Fixed the second parameter to be INamedTypeSymbol  Thus no overhead any more..